src/utils/database.ts

```diff
@@ -0,0 +1,24 @@
+import { Collection, InsertOneResult, MongoClient } from 'mongodb';
+import { Credential } from '../types';
+
+let client: MongoClient;
+
+export async function connectDatabase(url: string): Promise<void> {
+  client = new MongoClient(url);
+  await client.connect();
+}
+
+export function getCollection<T>(name: string): Collection<T> {
+  return client.db().collection<T>(name);
+}
+
+export function getCredentialCollection(): Collection<Credential> {
+  return getCollection<Credential>('credentials');
+}
+
+export async function addCredentialToDB(
+  credential: Credential
+): Promise<InsertOneResult<Credential>> {
+  const credentialCollection = await getCredentialCollection();
+  return credentialCollection.insertOne(credential);
+}
```

Comment:
I think this could be moved to another utility file, which would be more specific to credentials

Reply:
Oops, that was not the current state of my branch! So sorry! I deleted it later.